hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/webapp/dao/ContainersInfo.java

```diff
@@ -46,4 +47,7 @@ public ArrayList<ContainerInfo> getContainers() {
     return container;
   }
 
+  public void addAll(Collection<ContainerInfo> containersInfo) {
+    container.addAll(new ArrayList<>(containersInfo));
```

Comment:
You don't need to create the ArrayList.
```
public void addAll(Collection<ContainerInfo> containersInfo) {
   container.addAll(containersInfo);
}
```
Should just work: https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#addAll(java.util.Collection)

Reply:
Thanks for the suggestion, I will modify the code.